File: festo-dashboard-web/projects/festo-dashboard/src/lib/models/datasources/data-source.ts
import { Column } from './column';
export class DataSource {
    id: string;
    sourceId: string;
    connectionId: string;
    name: string;
    sourceType: string;
    schema: string;
    namespace: string;
    columns: Column[];
    parameters: Column[];
    selected = false;
    data: any[] = null;
}

File: API/Ngx.Dashboard/Ngx.Dashboard.Api/DataSources/Model/DataSource.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Ngx.Dashboard.Api.DataSources.Model
{
    public class DataSource
    {
        public string Id { set; get; }
        public string ConnectionId { set; get; }
        public string Name { set; get; }
        public string SourceType { set; get; }
        public string SourceId { set; get; }
        public string Schema { set; get; }
        public string Namespace { set; get; }
        public List<Column> Columns { set; get; }
        public List<Column> Parameters { set; get; }

    }
}
